File: LLM-Translations/GPT-4/avatar/Java/Python/codeforces_387_A.py
import datetime

def main():
    s = input().split(":")
    t = input().split(":")
    x = datetime.time(int(s[0]), int(s[1]))
    y = datetime.time(int(t[0]), int(t[1]))
    x_minutes = x.hour * 60 + x.minute
    y_minutes = y.hour * 60 + y.minute
    diff = x_minutes - y_minutes
    if diff < 0:
        diff += 24 * 60
    result = "{:02d}:{:02d}".format(diff // 60, diff % 60)
    print(result)

if __name__ == '__main__':
    main()

File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_387_A.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var h1, m1, h2, m2 int
	h1, m1 = getHourMinute() 
	m1 += h1 * 60
	h2, m2 = getHourMinute()
	m2 += h2 * 60
	m2 = m1 - m2
	m2 %= 1440
	fmt.Printf("%02d:%02d", m2 / 60, m2 % 60)
}

func getHourMinute() (int, int) {
	var input string
	fmt.Scanln(&input)
	parts := strings.Split(input, ":")
	hour, _ := strconv.Atoi(parts[0])
	minute, _ := strconv.Atoi(parts[1])
	return hour, minute
}

// End of Code
